src/github.com/getlantern/lantern-android/client/client.go

```diff
@@ -85,50 +80,33 @@ func (client *MobileClient) ServeHTTP() {
 	go client.pollConfiguration()
 }
 
-func (client *MobileClient) recordAnalytics() {
-
-	sessionPayload := &analytics.Payload{
-		HitType:    analytics.EventType,
-		Hostname:   "localhost",
-		TrackingId: trackingCodes["FireTweet"],
-		Event: &analytics.Event{
-			Category: "Session",
-			Action:   "Start",
-			Label:    runtime.GOOS,
-		},
-		UserAgent: "FireTweet",
-	}
-
-	// Report analytics, proxying through the local client. Note this
-	// is a little unorthodox by Lantern standards because it doesn't
-	// pin the certificate of the cloud.yaml root CA, instead relying
-	// on the go defaults.
-	httpClient, err := util.HTTPClient("", client.Client.Addr)
-	if err != nil {
-		log.Errorf("Could not create HTTP client %v", err)
-	} else {
-		analytics.SessionEvent(httpClient, sessionPayload)
-	}
-}
-
 // updateConfig attempts to pull a configuration file from the network using
 // the client proxy itself.
 func (client *MobileClient) updateConfig() error {
 	var buf []byte
 	var err error
+
 	if buf, err = pullConfigFile(client.fronter); err != nil {
 		log.Errorf("Could not update config: '%v'", err)
 		return err
 	}
-	return clientConfig.updateFrom(buf)
+	if err = clientConfig.updateFrom(buf); err == nil {
+		// Configuration changed, lets reload.
+		err := globals.SetTrustedCAs(clientConfig.getTrustedCerts())
+		if err != nil {
+			log.Debugf("Unable to configure trusted CAs: %s", err)
+		}
+		hqfc := client.Configure(clientConfig.Client)
+		client.fronter = hqfc.NewDirectDomainFronter()
+	}
+	return err
 }
 
 // pollConfiguration periodically checks for updates in the cloud configuration
 // file.
 func (client *MobileClient) pollConfiguration() {
 
-	// initially poll the config immediately
-	pollTimer := time.NewTimer(time.Second)
+	pollTimer := time.NewTimer(cloudConfigPollInterval)
```

Comment:
Wait so you don't want to grab the network config right away anymore @atavism? What's the thinking there?